server/src/main/java/org/elasticsearch/script/CtxMap.java

```diff
@@ -121,28 +138,30 @@ public void clear() {
         for (String key : metadata.keySet()) {
             metadata.remove(key);
         }
+        // TODO: this is just bogus, there isn't any case where metadata won't trip a failure above?
         source.clear();
```

Comment:
In every context there is are some read-only metadata keys, so this will currently always trip.